website/docs/extensions/write/onboarding-workflow.md

```diff
@@ -0,0 +1,531 @@
+        "items": {
+            "type": "object",
+            "properties": {
+            "title": {
+                "type": "string"
+            },
+            "url": {
+                "type": "string"
+            },
+            "group": {
+                "type": "string"
+            }
+            },
+            "required": [
+            "title",
+            "url",
+            "group"
+            ]
+        }
+    },
+}
+```
+
+Adding a complete list in the `package.json` can be confusing, so a better approach is to use a context value
+
+```json
+"value": ":warnings[${onboardingContext:warningsMarkdown}]"
+```
+
+at runtime, `${onboardingContext:warningsMarkdown}` is replaced by the actual list filled in the backend
+
+```
+const warnings = [];
+...
+const warning = {
+    state: res.successful ? 'successful' : 'failed',
+    description: res.description,
+    docDescription: res.docLinksDescription,
+    docLinks: res.docLinks,
+    command: res.fixCommand,
+};
+warnings.push(warning);
+
+extensionApi.context.setValue('warningsMarkdown', warnings, 'onboarding');
+```
+
+![img6](img/warnings_micromark.png)
+
+The **highlight** and **when** properties are optional. They are used to change the background color or define when the content column should be visible, respectively.
+
+#### Component
+
+Podman Desktop has some built-in components that can fit perfectly into an onboarding workflow, like the `create new connection` wizard.
+If you are working on an extension that allows creating a Kubernetes cluster, it would not make sense to recreate a page where the user can add the name, the resources to use, etc. This is when the component field comes in handy.
+
+By specifying the component you want to import, all the elements, styling, and actions are embedded into the step.
+
+Currently, Podman Desktop only supports two types of components for onboarding: `createContainerProviderConnection` and `createKubernetesProviderConnection`.
+
+An example can be seen in the Podman extension, where you can create a Podman machine during the workflow.
+
+```json
+{
+    "id": "createPodmanMachine",
+    "title": "Create a Podman machine",
+    "when": "!onboardingContext:podmanMachineExists && !isLinux",
+    "completionEvents": [
+        "onboardingContext:podmanMachineExists"
+    ],
+    "component": "createContainerProviderConnection"
+},
+```
+
+![img6](img/component_field.png)
+
+**N.B:** when using the **component** field, you should omit the **content**
+
+#### When
+
+The **when** property defines when a step must be visible. You can use any when clause, and Podman Desktop will evaluate it any time the context changes.
+
+#### State
+
+The **state**, when set, allows Podman Desktop to distinguish a normal step from a special one. It is used to associate a step with a failed state (`failed`) or, alternatively, with a complete state (`completed`).
+
+**N.B:** the last workflow step should have `completed` state.
```

Comment:
Actually one question... what's N.B.?

Reply:
lol, it's short for Nota Bene (note well): https://en.wikipedia.org/wiki/Nota_bene

Since it is more likely to mean New Brunswick here maybe we can just use 'Note:'?

Reply:
i thought it was used in english as well lol.
When writing in italian it's pretty common 😆